crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -202,6 +204,21 @@ impl<T: TransportConnect> Scheduler<T> {
         nodes_config: &NodesConfiguration,
         placement_hints: impl PartitionProcessorPlacementHints,
     ) -> Result<(), Error> {
+        // todo temporary band-aid to ensure convergence of multiple schedulers. Remove once we
+        //  accept equivalent configurations and remove persisting of the SchedulingPlan
+        if self.last_updated_scheduling_plan.elapsed() > Duration::from_secs(10) {
```

Comment:
Won't this mean that a change can go unnoticed if it was updated less that 10seconds ago. Maybe the fetch/assign can happen automatically on scheduler plan version change instead?

Reply:
The scheduler plan is not part of the metadata management. That's why we can't watch for version changes and then update (if I understood you correctly). We need to fetch it from the metadata store.

In case this `scheduler` instance tries to update the `SchedulingPlan`, then it will see the updated `SchedulingPlan`. This was already implemented. 

What this PR tries to solve is if there is a scheduler that is still operating on a older `SchedulingPlan` version but sees no need to update it (e.g. because all the nodes that it had assigned are alive). If this older `SchedulingPlan` is different than the newer one, then there can be contradicting instructions being sent to the `PPMs`. That's what this commit tries to solve by eventually bringing all nodes up to date wrt the `SchedulingPlan` version. It's true that there is a window of 10s in which we don't see updates if we are still fine with our current `SchedulingPlan` and see no need to update it.

Reply:
Ah sorry, I was under the wrong impression that anything stored in the Metadata is managed by default.

Thank you for the clarification